### PR TITLE
initial Triqler export

### PR DESCRIPTION
**Pull request by timosachsenberg**
# Description

initial Triqler export

# Checklist:
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

### CODE REVIEW

File: src/utils/TriqlerConverter.cpp [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        OPENMS_LOG_FATAL_ERROR << "FATAL: " << message << std::endl;
+        throw exit_code;
+      }
+    }
+};
+
+const String TOPPTriqlerConverter::param_in = "in";
+const String TOPPTriqlerConverter::param_in_design = "in_design";
+const String TOPPTriqlerConverter::param_method = "method";
+const String TOPPTriqlerConverter::param_Triqler_bioreplicate = "Triqler_bioreplicate";
```
Comment (timosachsenberg) [inline src/utils/TriqlerConverter.cpp:195]: todo: remove

File: src/utils/TriqlerConverter.cpp [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+const String TOPPTriqlerConverter::param_Triqler_condition = "Triqler_condition";
+const String TOPPTriqlerConverter::param_Triqler_mixture = "Triqler_mixture";
```
Comment (timosachsenberg) [inline src/utils/TriqlerConverter.cpp:197]: todo remove

File: src/tests/class_tests/openms/source/TriqlerFile_test.cpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+START_TEST(MSstatsFile, "$Id$")
+
+void OpenMS::TriqlerFile::storeLFQ(const String& filename,
+                                   const ConsensusMap& consensus_map,
+                                   const ExperimentalDesign& design,
+                                   const StringList& reannotate_filenames,
+                                   const String& condition,
+                                   const String& retention_time_summarization_method)
+
+START_SECTION(void OpenMS::MSstatsFile::storeLFQ( const OpenMS::String &filename,
```
Comment (timosachsenberg) [inline src/tests/class_tests/openms/source/TriqlerFile_test.cpp:51]: todo: rename

File: src/openms/source/FORMAT/TriqlerFile.cpp [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+                                   const ExperimentalDesign& design,
+                                   const StringList& reannotate_filenames,
+                                   const String& condition)
+{
+  // Experimental Design file
+  const ExperimentalDesign::SampleSection& sampleSection = design.getSampleSection();
+
+  if (design.getNumberOfLabels() != 1)
+  {
+     throw Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Too many labels for a label-free quantitation experiments. Please select the appropriate method, or validate the experimental desing.");
```
Comment (cbielow) [inline src/openms/source/FORMAT/TriqlerFile.cpp:127]: typo: `design`

File: src/utils/MSstatsConverter.cpp [hunk 5]
```diff
@@ -209,7 +206,6 @@ class TOPPMSstatsConverter final :
     static const String param_reannotate_filenames;
 
 private:
-
     static void fatalErrorIf_(const bool error_condition, const String &message, const int exit_code)
```
Comment (cbielow) [inline src/utils/MSstatsConverter.cpp:209]: move this to Exception.h and call it: throwReturnCode ...

question: if the exception is not caught, does the progam actually have the return code thrown?

File: src/openms/include/OpenMS/FORMAT/TriqlerFile.h [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+      };
+
+      /*
+        *  @brief Constructs the lines and adds them to the TextFile
+        *  @param peptideseq_quantifyable Has to be a set (only) for deterministic  ordered output
+        */
+      template <class LineType>
+      void constructFile_(TextFile& csv_out,
+                          const std::set<String>& peptideseq_quantifyable,
+                          LineType & peptideseq_to_prefix_to_intensities) const;
```
Comment (cbielow) [inline src/openms/include/OpenMS/FORMAT/TriqlerFile.h:204]: some formatting required here... there are some other occurrences of this

Comment (cbielow) [reply src/openms/include/OpenMS/FORMAT/TriqlerFile.h:204]: this is a template, but no definition. This is trouble.
But: it seems only one type is actually needed, so why not remove the template?

File: src/openms/include/OpenMS/FORMAT/TriqlerFile.h [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+        }
+        return result;
+      }
+
+      OpenMS::Peak2D::IntensityType meanIntensity_(const std::set< OpenMS::Peak2D::IntensityType > &intensities) const
+      {
+        return sumIntensity_(intensities) / intensities.size();
+      }
+
+      class TriqlerLine_
```
Comment (cbielow) [inline src/openms/include/OpenMS/FORMAT/TriqlerFile.h:135]: this class could be completely hidden in the implementation. Its not used externally, right?
Or would there be an advantage to let the world know that this exists?

File: src/utils/MSstatsConverter.cpp (hunk 5 above)
Comment (timosachsenberg) [reply src/utils/MSstatsConverter.cpp:209]: I have no idea ;) not my code

File: src/tests/topp/ProteomicsLFQ_2_out.consensusXML [hunk 8]
```diff
@@ -930,81 +921,91 @@
...
+			</PeptideIdentification>
+			<UserParam type="string" name="feature_id" value="3228988057475851378"/>
+		</consensusElement>
+		<consensusElement id="e_6577895467092461988" quality="0.417678" charge="2">
+			<centroid rt="1669.532387117258622" mz="508.247494448271027" it="1.592066e05"/>
 			<groupedElementList>
-				<element map="1" id="5529354363062563465" rt="1524.941088971155523" mz="508.247494448271027" it="2.140242e05" charge="2"/>
+				<element map="1" id="15960569366695886719" rt="1524.941088971155523" mz="508.247494448271027" it="2.140242e05" charge="2"/>
+				<element map="2" id="3485378768692219473" rt="1814.123685263361494" mz="508.247494448271027" it="1.04389e05" charge="2"/>
 			</groupedElementList>
```
Comment (timosachsenberg) [inline src/tests/topp/ProteomicsLFQ_2_out.consensusXML:958]: hier wurde es jetzt gelinkt

**Review by jpfeuffer — APPROVED** [review]

**Pull request merged**